lib/src/navbar/navbar-link/navbar-link.component.ts

```diff
@@ -1,20 +1,22 @@
-import { Component, OnInit, HostBinding, Input } from '@angular/core';
+import { Component, HostBinding, Input, ElementRef, ChangeDetectorRef } from '@angular/core';
 
 @Component({
     selector: 'hc-navbar-link',
     templateUrl: './navbar-link.component.html'
 })
-export class NavbarLinkComponent implements OnInit {
+export class NavbarLinkComponent {
     @HostBinding('class') hostClass = 'navbar-item';
     @Input() active?: boolean;
     @Input() uri: string;
     @Input() linkText: string;
     @Input() exact: boolean = false;
+    hidden: boolean = false;
 
-    constructor() {
+    constructor( private el: ElementRef, private ref: ChangeDetectorRef ) {
     }
 
-    ngOnInit() {
-    }
+    hide() { this.hidden = true; this.ref.detectChanges(); }
```

Comment:
Just curious why detectChanges() is needed here. Does it force the hide/show to happen faster so that the calculations in navResize() are accurate?